revapi-java/src/main/java-mr/9/org/revapi/java/compilation/MissingTypeAwareDelegatingElements.java

```diff
@@ -0,0 +1,132 @@
+
+import org.revapi.java.spi.IgnoreCompletionFailures;
+
+/**
+ * @author Lukas Krejci
+ *
+ * @since 0.1
+ */
+final class MissingTypeAwareDelegatingElements implements Elements {
+    private final Elements elements;
+
+    MissingTypeAwareDelegatingElements(Elements elements) {
+        this.elements = elements;
+    }
+
+    @Override
+    public PackageElement getPackageElement(CharSequence name) {
+        return IgnoreCompletionFailures.in(elements::getPackageElement, name);
+    }
+
+    @Override
+    public TypeElement getTypeElement(CharSequence name) {
+        return IgnoreCompletionFailures.in(elements::getTypeElement, name);
+    }
+
+    @Override
+    public Map<? extends ExecutableElement, ? extends AnnotationValue> getElementValuesWithDefaults(
+            AnnotationMirror a) {
+
+        return IgnoreCompletionFailures.in(elements::getElementValuesWithDefaults, a);
+    }
+
+    @Override
+    public String getDocComment(Element e) {
+        return IgnoreCompletionFailures.in(elements::getDocComment, e);
+    }
+
+    @Override
+    public boolean isDeprecated(Element e) {
+        return IgnoreCompletionFailures.in(elements::isDeprecated, e);
+    }
+
+    @Override
+    public Name getBinaryName(TypeElement type) {
+        return IgnoreCompletionFailures.in(elements::getBinaryName, type);
+    }
+
+    @Override
+    public PackageElement getPackageOf(Element type) {
+        return IgnoreCompletionFailures.in(elements::getPackageOf, type);
+    }
+
+    @Override
+    public List<? extends Element> getAllMembers(TypeElement type) {
+        return IgnoreCompletionFailures.in(elements::getAllMembers, type);
+    }
+
+    @Override
+    public List<? extends AnnotationMirror> getAllAnnotationMirrors(Element e) {
+        return IgnoreCompletionFailures.in(elements::getAllAnnotationMirrors, e);
+    }
+
+    @Override
+    public boolean hides(Element hider, Element hidden) {
+        return IgnoreCompletionFailures.in(elements::hides, hider, hidden);
+    }
+
+    @Override
+    public boolean overrides(ExecutableElement overrider, ExecutableElement overridden, TypeElement type) {
+        return IgnoreCompletionFailures.in(elements::overrides, overrider, overridden, type);
+    }
+
+    @Override
+    public String getConstantExpression(Object value) {
+        return IgnoreCompletionFailures.in(elements::getConstantExpression, value);
+    }
+
+    @Override
+    public void printElements(Writer w, Element... elems) {
+        IgnoreCompletionFailures.inVoid(elements::printElements, w, elems);
+    }
+
+    @Override
+    public boolean isFunctionalInterface(TypeElement type) {
+        return elements.isFunctionalInterface(type);
+    }
+
+    @Override
+    public Name getName(CharSequence cs) {
+        try {
+            return elements.getName(cs);
+        } catch (Exception ex) {
+            throw new IllegalStateException(ex);
+        }
+    }
+
+    @Override
+    public ModuleElement getModuleOf(Element e) {
```

Comment:
Does this work? According to the spec, the public API of the classes across all the versions must be exactly the same and here we're adding a new method. See https://docs.oracle.com/en/java/javase/23/docs/specs/jar/jar.html#multi-release-jar-files.

Reply:
This actually works (we are being able to build and use both java 8 and java 9 versions). Both tooling and runtime are not making any extensive API check so this can compile and run. Even when you add a class that does not supersede any class of the original code it still compile and run, son I assume no checks at all actually. 

Given that the one extending it's API is a JDK class here we have no option but to implement it this way or thinking about a bigger refactor. Let me think about this and I'll let you know.

Reply:
@metlos Do you consider the release of a major version that drops java 8 support an option? Going that way we could evaluate more additions too (I will still try an alternate implementation that respects the "same API" specification of the mrjars).